Src/FluentAssertions/Equivalency/DataTableEquivalencyStep.cs

```diff
@@ -146,46 +141,43 @@ private static void CompareCollections(IEquivalencyValidationContext context, IE
             {
                 if (selectedMembers.TryGetValue(collectionName, out IMember expectationMember))
                 {
-                    IMember matchingMember = FindMatchFor(expectationMember, context, config);
-
+                    IMember matchingMember = FindMatchFor(expectationMember, comparands.Subject, context.CurrentNode, config);
                     if (matchingMember is not null)
                     {
-                        IEquivalencyValidationContext nestedContext =
-                                context.AsNestedMember(expectationMember, matchingMember);
+                        IEquivalencyValidationContext nestedContext = context.AsNestedMember(expectationMember);
 
-                        if (nestedContext is not null)
```

Comment:
I assume the removed null check is because `AsNestedMember` never returns null?

Reply:
Yes. and it never returned `nulls`, so not sure how all these superfluous tests ended up in our code base.